src/main.rs

```diff
@@ -1,120 +1,116 @@
 use chrono::{NaiveDate, NaiveDateTime};
+use clap::Parser;
 use comrak::{markdown_to_html, ComrakOptions};
 use frontmatter_gen::{extract, Frontmatter, Value};
+use fs_extra::dir::{copy, CopyOptions};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use std::fs;
 use std::io;
-use std::path::{Path, PathBuf};
+use std::path::Path;
 use std::process;
+use std::sync::Arc;
 use tera::{Context, Tera};
 use walkdir::WalkDir;
 
+mod cli; // Import the CLI module
+mod server; // Import the server module
+
 fn main() -> io::Result<()> {
-    // Argument Parsing
-    let args: Vec<String> = std::env::args().collect();
-    if args.len() < 2 {
-        eprintln!("Usage: program <folder>");
-        process::exit(1);
-    }
-    let folder = PathBuf::from(&args[1]);
+    let args = cli::Cli::parse();
+
+    let input_folder = args.input_folder;
+    let output_folder = Arc::new(args.output_folder); // Convertemos para Arc<PathBuf>
+    let serve = args.serve;
+    let config_path = input_folder.join(args.config);
 
     // Initialize site data
-    let marmite = match fs::read_to_string("marmite.yaml") {
-        Ok(data) => data,
-        Err(e) => {
-            eprintln!("Unable to read marmite.yaml: {}", e);
-            process::exit(1);
-        }
-    };
+    let marmite = fs::read_to_string(config_path).unwrap_or_else(|e| {
+        eprintln!("Unable to read config file: {}", e);
+        process::exit(1);
+    });
+
     let site: Marmite = match serde_yaml::from_str(&marmite) {
         Ok(site) => site,
         Err(e) => {
             eprintln!("Failed to parse YAML: {}", e);
             process::exit(1);
         }
     };
+
     let mut site_data = SiteData::new(&site);
 
     // Walk through the content directory
-    for entry in WalkDir::new(folder.join(site_data.site.content_path)) {
-        match entry {
-            Ok(entry) => {
-                let path = entry.path();
-                if path.is_file() && path.extension().and_then(|e| e.to_str()) == Some("md") {
-                    if let Err(e) = process_file(path, &mut site_data) {
-                        eprintln!("Failed to process file {}: {}", path.display(), e);
-                    }
-                }
+    WalkDir::new(input_folder.join(site_data.site.content_path))
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| {
+            e.path().is_file() && e.path().extension().and_then(|ext| ext.to_str()) == Some("md")
+        })
+        .for_each(|entry| {
+            if let Err(e) = process_file(entry.path(), &mut site_data) {
+                eprintln!("Failed to process file {}: {}", entry.path().display(), e);
             }
-            Err(e) => eprintln!("Error reading entry: {}", e),
-        }
-    }
+        });
```

Comment:
Here, I did a small refactor and replaced a traditional `match` block with a more functional chain of methods that operate directly on the iterator. The new version uses `into_iter()` and `filter_map(Result::ok)` to filter valid entries, eliminating the need for an explicit `match` and improving code readability. Additionally, the `filter()` function checks if the file extensions are "md" in a more concise and straightforward way. 

The use of `for_each()` to iterate and process the files also reduces boilerplate code, making the flow smoother and easier to understand. Overall, this approach feels more idiomatic in Rust.